Add properties and substation properties fields in expert filters.

FranckLecuyer opened the pull request:
Waiting for next gridsuite-filter release ...

FranckLecuyer committed:
Add properties and substation properties fields in expert filters.

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Merge branch 'main' into add_properties_filtering_in_expert_filter

FranckLecuyer committed:
Fix eslint errors

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Change name of autocomplete component for property name

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Change function into callback

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

thangqp commented on src/components/dialogs/filter/expert/expert-filter-utils.ts:
```diff
@@ -56,6 +56,14 @@ const getDataType = (fieldName: string, operator: string) => {
+        fieldName === FieldType.SUBSTATION_PROPERTY_2
+    ) {
+        return DataType.PROPERTY;
+    }
```
do not need this check particularly, getting dataType has been implemented in the below code

FranckLecuyer replied:
Removed

FranckLecuyer committed:
Merge branch 'main' into add_properties_filtering_in_expert_filter

FranckLecuyer committed:
Fix pb with uncontrolled to controlled component

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Change to have for property fields :

<property> 'is' <property name> <operator> <property values>

where : <property> is the field name 'Property' or 'Substation property'
<property name> is the name of the choosen equipment property
<property value> is a list of property values
<operator> is the operator (just 'in' for the moment) used to check the <property name> against the <property values>

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Merge branch 'main' into add_properties_filtering_in_expert_filter

FranckLecuyer committed:
Fix unconsidered property value deletion

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Remove commented line

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Remove useless line

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Fix pbs in PropertyValueEditor component, by replacing the use of rhf components by the use of material ui components.

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

thangqp commented on src/components/utils/rqb-inputs/property-value-editor.tsx:
```diff
@@ -0,0 +1,131 @@
+        <Grid container item spacing={1} columns={100}>
+            <Grid item xs={40}>
+                <Autocomplete
+                    value={propertyName}
```
propertyName ?? '' to avoid error uncontrolled to controlled in console

FranckLecuyer replied:
OK

thangqp commented on src/components/utils/rqb-inputs/property-value-editor.tsx:
```diff
@@ -0,0 +1,131 @@
+    }, [equipmentType, setEquipmentType]);
+
+    const predefinedNames = useMemo(() => {
+        return Object.keys(equipmentPredefinedProps ?? []).sort();
```
{} empty object instead of empty array

FranckLecuyer replied:
OK

thangqp commented on src/components/utils/rqb-inputs/property-value-editor.tsx:
```diff
@@ -0,0 +1,131 @@
+    );
+
+    return (
+        <Grid container item spacing={1} columns={100}>
```
default number of columns in Grid system is 12, why need to redefine as 100?

FranckLecuyer replied:
Changed

thangqp commented on src/components/utils/rqb-inputs/property-value-editor.tsx:
```diff
@@ -0,0 +1,131 @@
+                >
+                    {Object.values(PROPERTY_VALUE_OPERATORS).map((operator) => (
+                        <MenuItem key={operator.id} value={operator.id}>
+                            {operator.label}
```
lack of translation

FranckLecuyer replied:
Fixed

FranckLecuyer committed:
Merge branch 'main' into add_properties_filtering_in_expert_filter

FranckLecuyer committed:
Changes after review

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Avoid duplicating operators

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Remove useless constant

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>

FranckLecuyer committed:
Remove Object.values

Signed-off-by: Franck LECUYER <franck.lecuyer@rte-france.com>
